opencloudComLibrary/src/main/java/eu/opencloud/android/lib/resources/files/tus/CreateTusUploadRemoteOperation.kt

```diff
@@ -0,0 +1,199 @@
+package eu.opencloud.android.lib.resources.files.tus
+
+import android.util.Base64
+import eu.opencloud.android.lib.common.OpenCloudClient
+import eu.opencloud.android.lib.common.http.HttpConstants
+import eu.opencloud.android.lib.common.http.methods.nonwebdav.HeadMethod
+import eu.opencloud.android.lib.common.http.methods.nonwebdav.PostMethod
+import eu.opencloud.android.lib.common.network.ChunkFromFileRequestBody
+import eu.opencloud.android.lib.common.network.WebdavUtils
+import eu.opencloud.android.lib.common.operations.RemoteOperation
+import eu.opencloud.android.lib.common.operations.RemoteOperationResult
+import eu.opencloud.android.lib.common.operations.RemoteOperationResult.ResultCode
+import eu.opencloud.android.lib.common.utils.isOneOf
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import timber.log.Timber
+import java.io.File
+import java.io.FileInputStream
+import java.io.RandomAccessFile
+import java.net.URL
+import java.nio.channels.FileChannel
+import java.security.MessageDigest
+
+class CreateTusUploadRemoteOperation(
+    private val file: File,
+    private val remotePath: String,
+    private val mimetype: String,
+    private val metadata: Map<String, String>,
+    private val useCreationWithUpload: Boolean,
+    private val firstChunkSize: Long?,
+    private val tusUrl: String,
+    private val collectionUrlOverride: String? = null,
+) : RemoteOperation<String>() {
+
+    override fun run(client: OpenCloudClient): RemoteOperationResult<String> {
+        return try {
+            // Determine TUS endpoint URL based on provided parameters
+            val targetFileUrl = when {
+                // 1. Use collectionUrlOverride if provided (for TUS support checks)
+                !collectionUrlOverride.isNullOrBlank() -> collectionUrlOverride
+                
+                // 2. Use tusUrl if provided and not empty (for existing upload sessions)
+                !tusUrl.isNullOrBlank() -> tusUrl
+                
+                // 3. Construct TUS endpoint URL to parent directory (server expects directory, not file)
+                else -> {
+                    val parentDir = remotePath.substringBeforeLast('/', "/")
+                    buildString {
+                        append(client.baseUri.toString().trimEnd('/'))
+                        append("/remote.php/webdav")
+                        append(WebdavUtils.encodePath(parentDir))
+                        if (!endsWith("/")) append("/")
+                    }
+                }
+            }
+
+            Timber.d("TUS Creation URL: %s", targetFileUrl)
+            
+            // TUS Upload-Metadata (decoded for debugging)
+            val filename = remotePath.substringAfterLast('/')
+            val mtime = (file.lastModified() / 1000).toString()
+            val checksum = "sha1 " + computeSha1(file)
```

Comment:
Why is there a sha1 computed here?
Is the checksum (sha256) not coming from the caller of this function?
Maybe I'm misunderstanding something